src/services/questions.rs

```diff
@@ -0,0 +1,82 @@
+use std::sync::Arc;
+use utoipa_axum::{router::OpenApiRouter, routes};
+use crate::server::AppState;
+use std::{collections::HashSet, ops::Deref};
+use axum::{extract::State, Json};
+use bedrock::packet::{Problem, Test};
+
+#[derive(serde::Serialize, utoipa::ToSchema)]
+pub struct TestResponse {
+    pub input: String,
+    pub output: String,
+}
+
+impl From<&Test> for TestResponse {
+    fn from(value: &Test) -> Self {
+        Self {
+            input: value.input.clone(),
+            output: value.output.clone(),
+        }
+    }
+}
+
+#[derive(serde::Serialize, utoipa::ToSchema)]
+pub struct QuestionResponse {
+    languages: Option<HashSet<String>>,
+    title: String,
+    description: Option<String>,
+    tests: Vec<TestResponse>,
+}
+
+impl From<&Problem> for QuestionResponse {
+    fn from(value: &Problem) -> Self {
+        Self {
+            languages: value.languages.clone(),
+            title: value.title.clone(),
+            description: value.description.as_ref().map(|x| x.html().unwrap()),
+            tests: value.tests.iter().map(From::from).collect(),
+        }
+    }
+}
+
+#[derive(serde::Serialize, utoipa::ToSchema)]
+pub struct AllQuestionResponse(Vec<QuestionResponse>);
+
+#[axum::debug_handler]
+#[utoipa::path(get, path = "/", responses((status = OK, body = AllQuestionResponse, content_type = "application/json")))]
+pub async fn get_all(State(state): State<Arc<AppState>>) -> Json<AllQuestionResponse> {
+    let questions = state
+        .config
+        .packet
+        .problems
+        .iter()
+        .map(|x| x.deref().into())
+        .collect();
+
+    Json(AllQuestionResponse(questions))
+}
+
+#[axum::debug_handler]
+#[utoipa::path(get, path = "/{id}", responses((status = OK, body = QuestionResponse, content_type = "application/json"),(status=40r, description="Question Not Found"),))]
```

Comment:
`40r` -> `404` and format to be consistent with auth
```suggestion
#[utoipa::path(
    get,
    path = "/{id}",
    responses(
        (status=OK, body=QuestionResponse, content_type="application/json"),
        (status=404, description="Question Not Found"),
    )
)]
```